pennylane/ops/qubit/arithmetic_ops.py

```diff
@@ -118,12 +118,12 @@ def _matrix(cls, *params):
         return QubitCarry._mat
 
     @staticmethod
-    def decomposition(wires):
-        decomp_ops = [
+    def compute_decomposition(wires):
+        decomp_ops = (
             qml.Toffoli(wires=wires[1:]),
             qml.CNOT(wires=[wires[1], wires[2]]),
             qml.Toffoli(wires=[wires[0], wires[2], wires[3]]),
-        ]
+        )
```

Comment:
It seems a good choice, but is there any deeper reason for a tuple over a list here?

Reply:
Prevents modification of it later, so it's a little safer if we don't intend for the object to be modified.

If the iterable isn't going to be modified down the line, tuples are slightly more efficient, because they don't need to support the overhead of resizing.

Reply:
This may be a breaking change all by itself 😆 I worry that this might cause issues for developers in that all `Sequence[Operations]` in the codebase (I believe) are currently lists?